src/track/beats.h

```diff
@@ -54,124 +84,264 @@ class Beats : public QObject {
-
-    // Starting from sample dSamples, return the sample of the closest beat in
-    // the track, or -1 if none exists.  Non- -1 values are guaranteed to be
-    // even.
-    virtual double findClosestBeat(double dSamples) const = 0;
-
-    // Find the Nth beat from sample dSamples. Works with both positive and
-    // negative values of n. Calling findNthBeat with n=0 is invalid. Calling
-    // findNthBeat with n=1 or n=-1 is equivalent to calling findNextBeat and
-    // findPrevBeat, respectively. If dSamples refers to the location of a beat,
-    // then dSamples is returned. If no beat can be found, returns -1.
-    virtual double findNthBeat(double dSamples, int n) const = 0;
-
-    int numBeatsInRange(double dStartSample, double dEndSample);
-
-    // Find the sample N beats away from dSample. The number of beats may be
-    // negative and does not need to be an integer.
-    double findNBeatsFromSample(double fromSample, double beats) const;
-
-
-    // Adds to pBeatsList the position in samples of every beat occurring between
-    // startPosition and endPosition. BeatIterator must be iterated while
-    // holding a strong references to the Beats object to ensure that the Beats
-    // object is not deleted. Caller takes ownership of the returned BeatIterator;
-    virtual std::unique_ptr<BeatIterator> findBeats(double startSample, double stopSample) const = 0;
-
-    // Return whether or not a sample lies between startPosition and endPosition
-    virtual bool hasBeatInRange(double startSample, double stopSample) const = 0;
-
-    // Return the average BPM over the entire track if the BPM is
-    // valid, otherwise returns -1
-    virtual double getBpm() const = 0;
-
-    // Return the average BPM over the range from startSample to endSample,
-    // specified in samples if the BPM is valid, otherwise returns -1
-    virtual double getBpmRange(double startSample, double stopSample) const = 0;
-
-    // Return the average BPM over the range of n*2 beats centered around
-    // curSample.  (An n of 4 results in an averaging of 8 beats).  Invalid
-    // BPM returns -1.
-    virtual double getBpmAroundPosition(double curSample, int n) const = 0;
+    /// Starting from frame, return the frame number of the next beat
+    /// in the track, or -1 if none exists. If frame refers to the location
+    /// of a beat, frame is returned.
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual double findNextBeat(double sample) const {
+        return findNextBeatNew(sample / 2.0) * 2.0;
+    }
+    virtual FrameNum findNextBeatNew(FrameNum frame) const;
+
+    /// Starting from frame frame, return the frame number of the previous
+    /// beat in the track, or -1 if none exists. If frame refers to the
+    /// location of beat, frame is returned.
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual double findPrevBeat(FrameNum frame) const {
```

Comment:
I am confused. Isn't this the legacy function that takes a sample position? If so, it shouldn't take a FrameNum type!

Reply:
If implicit conversions are possible between frame and sample types, that defeats the purpose of having types for each.

Reply:
@Be-ing should I create a new class for Frame to ensure type safety? We won't have a sample type or class since we'll never operate on samples.

Reply:
Yes